src/seedu/addressbook/commands/Command.java

```diff
@@ -54,6 +54,17 @@ public void setData(AddressBook addressBook, List<? extends ReadOnlyPerson> rele
         this.relevantPersons = relevantPersons;
     }
 
+    /**
+     * Returns true if the command executed mutates the actual data
+     * By default set to return false assuming all commands do not mutate the actual data
+     * Commands that actually mutates the data - AddCommand, ClearCommand, DeleteCommand
+     *
+     * @return false
+     */
+    public boolean isMutating() {
+        return false;
```

Comment:
Having a default implementation reduces the amount of code, but might make the code less intuitive. Making the code `abstract` would force all the children class to declare whether it is a mutating command or not, instead of depending on arbitrary decisions on the parent class.